### ISSUE TITLE
PHP 8.X and Laravel 11 suport

### ISSUE THREAD

**Issue by ven0ms99** [beyondcode/laravel-er-diagram-generator#112]
```
composer require beyondcode/laravel-er-diagram-generator --dev
Using version ^3.0 for beyondcode/laravel-er-diagram-generator
./composer.json has been updated
Running composer update beyondcode/laravel-er-diagram-generator
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/common[v2.4.0, ..., v2.5.3] require doctrine/inflector 1.* -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/common[v2.6.0, ..., v2.6.2] require php ~5.5|~7.0 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/common[v2.7.0, ..., v2.7.3] require php ~5.6|~7.0 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/common[v2.8.0, ..., v2.8.1] require php ~7.1 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/common[v2.9.0, ..., 2.13.0] require php ^7.1 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/dbal[v2.6.0, ..., v2.9.3] require php ^7.1 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/common[2.13.1, ..., 2.13.3] require doctrine/inflector ^1.0 -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/dbal[v2.10.0, ..., 2.10.4] require php ^7.2 -> your php version (8.3.4) does not satisfy that requirement.
    - doctrine/dbal[2.11.0, ..., 2.11.3] require php ^7.3 -> your php version (8.3.4) does not satisfy that requirement.
    - laravel/framework is locked to version v11.0.5 and an update of this package was not requested.
    - carbonphp/carbon-doctrine-types 3.2.0 conflicts with doctrine/dbal 3.8.3.
    - beyondcode/laravel-er-diagram-generator 3.0.0 requires doctrine/dbal ~2.3|^3.3 -> satisfiable by doctrine/dbal[2.3.0, ..., 2.13.9, 3.3.0, ..., 3.8.3].
    - nesbot/carbon 3.1.1 requires carbonphp/carbon-doctrine-types * -> satisfiable by carbonphp/carbon-doctrine-types[3.2.0].
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.5.4.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.4.6.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.3.8.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.3.5.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.9.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.8.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.1.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.12.1.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.3.5.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.3.2.
    - doctrine/dbal[v2.5.0, ..., v2.5.2] require doctrine/common >=2.4,<2.6-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.5.3].
    - doctrine/dbal[v2.5.3, ..., v2.5.5] require doctrine/common >=2.4,<2.7-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.6.2].
    - doctrine/dbal[v2.5.6, ..., v2.5.13] require doctrine/common >=2.4,<2.8-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.7.3].
    - doctrine/dbal[v2.4.0, ..., v2.4.5] require doctrine/common ~2.4 -> satisfiable by doctrine/common[v2.4.0, ..., 2.13.3].
    - laravel/framework v11.0.5 requires nesbot/carbon ^2.72.2|^3.0 -> satisfiable by nesbot/carbon[3.1.1].
    - laravel/framework v11.0.5 requires symfony/http-foundation ^7.0 -> satisfiable by symfony/http-foundation[v7.0.4].
    - Root composer.json requires beyondcode/laravel-er-diagram-generator ^3.0 -> satisfiable by beyondcode/laravel-er-diagram-generator[3.0.0].
```

**Comment by w3spi5** [comment]
I've used it on Laravel 11 PHP 8.2.16 without problem, but result is strange, I think an improvement is required

**Comment by krebbi** [comment]
I'm having the same issue with Laravel 11 & PHP 8.2.2

./composer.json has been updated
Running composer update beyondcode/laravel-er-diagram-generator
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/dbal[v2.5.0, ..., v2.5.2] require doctrine/common >=2.4,<2.6-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.5.3].
    - doctrine/dbal[v2.5.3, ..., v2.5.5] require doctrine/common >=2.4,<2.7-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.6.2].
    - doctrine/dbal[v2.5.6, ..., v2.5.13] require doctrine/common >=2.4,<2.8-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.7.3].
    - doctrine/dbal[v2.4.0, ..., v2.4.5] require doctrine/common ~2.4 -> satisfiable by doctrine/common[v2.4.0, ..., 2.13.3].
    - laravel/framework v11.0.7 requires nesbot/carbon ^2.72.2|^3.0 -> satisfiable by nesbot/carbon[3.1.1].
    - laravel/framework v11.0.7 requires symfony/http-foundation ^7.0 -> satisfiable by symfony/http-foundation[v7.0.4].
    - beyondcode/laravel-er-diagram-generator[0.1.0, ..., 0.2.3, 1.0.0, ..., 1.5.0] require php ^7.1 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/common[v2.9.0, ..., 2.13.0] require php ^7.1 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/dbal[v2.6.0, ..., v2.9.3] require php ^7.1 -> your php version (8.2.2) does not satisfy that requirement.
    - beyondcode/laravel-er-diagram-generator 2.0.0 requires nikic/php-parser ^2.0|^3.0|^4.0 -> found nikic/php-parser[v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, ..., v3.1.5, v4.0.0, ..., v4.19.1] but the package is fixed to v5.0.2 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/common[v2.4.0, ..., v2.5.3] require doctrine/inflector 1.* -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/common[v2.6.0, ..., v2.6.2] require php ~5.5|~7.0 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/common[v2.7.0, ..., v2.7.3] require php ~5.6|~7.0 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/common[v2.8.0, ..., v2.8.1] require php ~7.1 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/common[2.13.1, ..., 2.13.3] require doctrine/inflector ^1.0 -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/dbal[v2.10.0, ..., 2.10.4] require php ^7.2 -> your php version (8.2.2) does not satisfy that requirement.
    - doctrine/dbal[2.11.0, ..., 2.11.3] require php ^7.3 -> your php version (8.2.2) does not satisfy that requirement.
    - laravel/framework is locked to version v11.0.7 and an update of this package was not requested.
    - carbonphp/carbon-doctrine-types 3.2.0 conflicts with doctrine/dbal 3.8.3.
    - beyondcode/laravel-er-diagram-generator 3.0.0 requires doctrine/dbal ~2.3|^3.3 -> satisfiable by doctrine/dbal[2.3.0, ..., 2.13.9, 3.3.0, ..., 3.8.3].
    - nesbot/carbon 3.1.1 requires carbonphp/carbon-doctrine-types * -> satisfiable by carbonphp/carbon-doctrine-types[3.2.0].
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.5.4.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.4.6.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.3.8.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 3.3.5.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.9.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.8.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.13.1.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.12.1.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.3.5.
    - symfony/http-foundation v7.0.4 conflicts with doctrine/dbal 2.3.2.
    - Root composer.json requires beyondcode/laravel-er-diagram-generator * -> satisfiable by beyondcode/laravel-er-diagram-generator[0.1.0, ..., 0.2.3, 1.0.0, ..., 1.5.0, 2.0.0, 3.0.0].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require beyondcode/laravel-er-diagram-generator:*" to figure out if any version is installable, or "composer require beyondcode/laravel-er-diagram-generator:^2.1" if you know which you need.

**Comment by noud** [comment]
yes, i have same issue, Laravel is too new.

Can someone make a new distribution with these patches to composer.json

-        "php": "^7.1|^8.0",
+        "php": "^7.1|^8.0|^8.1|^8.2",

-        "doctrine/dbal": "~2.3|^3.3",
+        "doctrine/dbal": "~2.3|^3.3|^3.8|^4.0",

-        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10",
+        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10|^10.5.20|^11.1.3",

thanks,
Noud

**Comment by noud** [comment]
yes, same issue, Laravel is too new.

Can you make a new distribution with these patches to composer.json
```
-        "php": "^7.1|^8.0",
+        "php": "^7.1|^8.0|^8.1|^8.2",

-        "doctrine/dbal": "~2.3|^3.3",
+        "doctrine/dbal": "~2.3|^3.3|^3.8|^4.0",

-        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10",
+        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10|^10.5.20|^11.1.3",
```

thanks,
Noud

**Comment by Semen12** [comment]
> yes, same issue, Laravel is too new.
> 
> Can you make a new distribution with these patches to composer.json
> 
> ```
> -        "php": "^7.1|^8.0",
> +        "php": "^7.1|^8.0|^8.1|^8.2",
> 
> -        "doctrine/dbal": "~2.3|^3.3",
> +        "doctrine/dbal": "~2.3|^3.3|^3.8|^4.0",
> 
> -        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10",
> +        "phpunit/phpunit": "^7.0| ^8.0|^9.5.10|^10.5.20|^11.1.3",
> ```
> 
> thanks, Noud

it's now work (((

**Comment by oppytut** [comment]
I am trying to install on Laravel 11 & PHP 8.3.9
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - doctrine/dbal[v2.5.0, ..., v2.5.2] require doctrine/common >=2.4,<2.6-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.5.3].
    - doctrine/dbal[v2.5.3, ..., v2.5.5] require doctrine/common >=2.4,<2.7-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.6.2].
    - doctrine/dbal[v2.5.6, ..., v2.5.13] require doctrine/common >=2.4,<2.8-dev -> satisfiable by doctrine/common[v2.4.0, ..., v2.7.3].
    - doctrine/dbal[v2.4.0, ..., v2.4.5] require doctrine/common ~2.4 -> satisfiable by doctrine/common[v2.4.0, ..., 2.13.3].
    - laravel/framework v11.20.0 requires nesbot/carbon ^2.72.2|^3.0 -> satisfiable by nesbot/carbon[3.7.0].
    - laravel/framework v11.20.0 requires symfony/http-foundation ^7.0 -> satisfiable by symfony/http-foundation[v7.1.3].
    - beyondcode/laravel-er-diagram-generator[0.1.0, ..., 0.2.3, 1.0.0, ..., 1.5.0] require php ^7.1 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/common[v2.9.0, ..., 2.13.0] require php ^7.1 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/dbal[v2.6.0, ..., v2.9.3] require php ^7.1 -> your php version (8.3.9) does not satisfy that requirement.
    - beyondcode/laravel-er-diagram-generator 2.0.0 requires nikic/php-parser ^2.0|^3.0|^4.0 -> found nikic/php-parser[v2.0.0, v2.0.1, v2.1.0, v2.1.1, v3.0.0, ..., v3.1.5, v4.0.0, ..., v4.19.1] but the package is fixed to v5.1.0 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/common[v2.4.0, ..., v2.5.3] require doctrine/inflector 1.* -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/common[v2.6.0, ..., v2.6.2] require php ~5.5|~7.0 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/common[v2.7.0, ..., v2.7.3] require php ~5.6|~7.0 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/common[v2.8.0, ..., v2.8.1] require php ~7.1 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/common[2.13.1, ..., 2.13.3] require doctrine/inflector ^1.0 -> found doctrine/inflector[v1.0, ..., 1.4.4] but the package is fixed to 2.0.10 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - doctrine/dbal[v2.10.0, ..., 2.10.4] require php ^7.2 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/dbal[2.11.0, ..., 2.11.3] require php ^7.3 -> your php version (8.3.9) does not satisfy that requirement.
    - doctrine/dbal[2.13.0, ..., 2.13.8, 3.3.0, ..., 3.3.5] require doctrine/deprecations ^0.5.3 -> found doctrine/deprecations[v0.5.3] but the package is fixed to 1.1.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - laravel/framework is locked to version v11.20.0 and an update of this package was not requested.
    - carbonphp/carbon-doctrine-types 3.2.0 conflicts with doctrine/dbal 3.8.7.
    - beyondcode/laravel-er-diagram-generator 3.0.0 requires doctrine/dbal ~2.3|^3.3 -> satisfiable by doctrine/dbal[2.3.0, ..., 2.13.9, 3.3.0, ..., 3.8.7].
    - nesbot/carbon 3.7.0 requires carbonphp/carbon-doctrine-types * -> satisfiable by carbonphp/carbon-doctrine-types[3.2.0].
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 3.5.4.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 3.4.6.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 3.3.8.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 2.13.9.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 2.12.1.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 2.3.5.
    - symfony/http-foundation v7.1.3 conflicts with doctrine/dbal 2.3.2.
    - Root composer.json requires beyondcode/laravel-er-diagram-generator * -> satisfiable by beyondcode/laravel-er-diagram-generator[0.1.0, ..., 0.2.3, 1.0.0, ..., 1.5.0, 2.0.0, 3.0.0].
```
